wave/src/Http/Controllers/StripeController.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+namespace Wave\Http\Controllers;
+
+use Wave\Plan;
+use Wave\User;
+use Carbon\Carbon;
+use Illuminate\Support\Str;
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use Illuminate\Support\Facades\Auth;
+use App\Notifications\NewUserPassword;
+
+class StripeController extends Controller
+{
+
+    public function postCreateCheckoutSession(Request $request)
+    {
+
+        $stripe = new \Stripe\StripeClient(env('STRIPE_SECRET'));
+
+        $sessionData = [
+            'success_url' => route('stripe.success-checkout') . "?session_id={CHECKOUT_SESSION_ID}",
+            'cancel_url' => route('stripe.error-checkout'),
+            'line_items' => [
+                [
+                    'price' => $request->get('planId'),
+                    'quantity' => 1,
+                ],
+            ],
+            'mode' => 'subscription',
+            'tax_id_collection' => [
+                'enabled' => env('CASHIER_STRIPE_CALCULATE_TAXES')
+            ],
+            'allow_promotion_codes' => env('CASHIER_STRIPE_ALLOW_PROMO_CODES'),
+            'locale' => "auto",
+        ];
+
+        if (auth()->check() && auth()->user()->stripe_id) {
+            $sessionData['customer'] = auth()->user()->stripe_id;
+        }
+
+        if (auth()->check() && !auth()->user()->stripe_id) {
+            $sessionData['customer_email'] = auth()->user()->email;
+        }
+
+        $session = $stripe->checkout->sessions->create($sessionData);
+
+        return response()->json($session, 200);
+    }
+
+    /**
+     * Handle the success Stripe Checkout
+     */
+    public function postSuccessCheckout(Request $request)
+    {
+        if (!$request->get('session_id')) abort(401);
+        $stripe = new \Stripe\StripeClient(env('STRIPE_SECRET'));
+
+        $session = $stripe->checkout->sessions->retrieve($request->get('session_id'));
+        $customer = $stripe->customers->retrieve($session->customer);
+        if (!$session || !$customer) abort(401);
+
+        $subscription = $stripe->subscriptions->retrieve($session->subscription);
+
+        $user = $this->createUserFromCheckout($customer);
+        $this->createStripeSubscription($user, $subscription);
+
+        // Update User role
+        $plan = Plan::where('plan_id', $subscription->plan->id)->first();
+        // add associated role to user
+        $user->role_id = $plan->role_id;
+        $user->save();
+
+        // Authenticate the user
+        Auth::login($user);
+
+
+        session()->flash('complete');
+        return view('theme::welcome');
+    }
+
+    public function postErrorCheckout(Request $request)
+    {
+    }
+
+    private function createUserFromCheckout(\Stripe\Customer $customer)
+    {
+        $user = User::where('email', $customer->email)->first();
+        if ($user) {
+            if (!$user->stripe_id) {
+                $user->stripe_id = $customer->id;
+                $user->save();
+
+                $user->syncStripeCustomerDetails();;
```

Comment:
```suggestion
                $user->syncStripeCustomerDetails();
```

Found a straggling semi-colon left behind.